Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/AlpideCoder.h

```diff
@@ -445,6 +447,141 @@ class AlpideCoder
     return chipData.getData().size();
   }
 
+  /// Verifies the decoder by comparing the contents a cable by re-encoding seen
+  /// chips back into the ALPIDE format.
+  template <typename LG, typename CG>
+  static bool verifyDecodedCable(
+    std::map<int, ChipPixelData*>& seenChips, PayLoadCont& buffer,
+    std::vector<uint16_t>& seenChipIDs, LG lidGetter, CG cidGetter)
+  {
+    PayLoadCont reconstructedData;
+
+    // Ensure the length of the reconstructed buffer.
+    int bufferLength = 0;
+    for (auto it = seenChips.begin(); it != seenChips.end(); ++it) {
+      bufferLength += it->second->getData().size();
+    }
+    bufferLength += seenChipIDs.size() * 2;
+    reconstructedData.ensureFreeCapacity(40 * bufferLength);
+
+    // Encode the seen chips in the order they were decoded.
+    for (int ID : seenChipIDs) {
+      ChipPixelData currentChip;
```

Comment:
Yes, since in the case of an absence of the chip ID, we want to still encode an empty chip, so initializing an empty `ChipPixelData` on the stack is important.